common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -70,100 +73,93 @@ public static String modifyMultipleMatchKeyword(String query){
+                    final String keyword = modifyMultipleMatchKeyword(query);
 
+                    if(keyword != null){
                         conceptCodeIDName.add(criteriaBuilder.equal(root.get("conceptCode"),
                                 criteriaBuilder.literal(query)));
-
                         try {
                             long conceptId = Long.parseLong(query);
                             conceptCodeIDName.add(criteriaBuilder.equal(root.get("conceptId"),
                                     criteriaBuilder.literal(conceptId)));
                         } catch (NumberFormatException e) {
                             // Not a long, don't try to match it to a concept ID.
                         }
-
-
                         matchExp = criteriaBuilder.function("match", Double.class,
                                 root.get("conceptName"), criteriaBuilder.literal(keyword));
-
+                        matchSynonymExp = criteriaBuilder.function("match", Double.class,
+                                csJoin.get("conceptSynonymName"), criteriaBuilder.literal(keyword));
                     }
 
 
-                    // Optionally filter on standard concept, vocabulary ID, domain ID
                     if (standardConceptFilter.equals(StandardConceptFilter.STANDARD_CONCEPTS)) {
 
-                        if(!Strings.isNullOrEmpty(query)) {
+                        if(keyword != null) {
                             conceptCodeIDName.add(criteriaBuilder.greaterThan(matchExp, 0.0));
+                            conceptCodeIDName.add(criteriaBuilder.greaterThan(matchSynonymExp, 0.0));
                             predicates.add(
                                     criteriaBuilder.or(
                                             conceptCodeIDName.toArray(new Predicate[0])
                                     )
                             );
                         }
 
-                        List<Predicate> standardConceptPredicates = new ArrayList<>();
-                        standardConceptPredicates.add(criteriaBuilder.equal(root.get("standardConcept"),
-                                criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
-                        standardConceptPredicates.add(criteriaBuilder.equal(root.get("standardConcept"),
-                                criteriaBuilder.literal(CLASSIFICATION_CONCEPT_CODE)));
-
                         predicates.add(
                                 criteriaBuilder.or(standardConceptPredicates.toArray(new Predicate[0]))
                         );
 
-                    } else if (standardConceptFilter.equals(StandardConceptFilter.NON_STANDARD_CONCEPTS)) {
-
-                        if(!Strings.isNullOrEmpty(query)) {
+                    }else if (standardConceptFilter.equals(StandardConceptFilter.NON_STANDARD_CONCEPTS)) {
 
+                        if(keyword != null) {
                             conceptCodeIDName.add(criteriaBuilder.greaterThan(matchExp, 0.0));
```

Comment:
Is the reason that we need this that not all concepts have synonym rows? (Just confirming.)

Reply:
Do you mean to add the match query here ? We no longer do the synonym query outside this. We got the jpa criteria to work with join. 

But it is true that not all concepts have synonyms which is why we search concept_name or concep_synonym_name.

Reply:
Yeah, that's what I was checking -- thanks.